File: go/srv2.go
package main

import (
	"log"
	"net"
)

func main() {
	l, err := net.Listen("tcp", "127.0.0.1:7012")
	if err != nil {
		log.Fatal("Error listening: ", err)
	}
	defer l.Close()

	for {
		if conn, err := l.Accept(); err != nil {
			log.Fatal("Error accepting: ", err)
		} else {
			go handler(conn)
		}
	}
}

func handler(conn net.Conn) {
	defer conn.Close()
	buf := make([]byte, 1024)
	if _, err := conn.Read(buf); err != nil {
		// log.Fatal("Error reading:", err)
	}

	res := `HTTP/1.0 200 OK
Connection: keep-alive
Content-Length: 5
Content-Type: text/plain

Hello`
	conn.Write([]byte(res))
}

File: lua/srv2.lua
local server = require 'http.server'
local headers = require 'http.headers'

local s = server.listen {
  host = 'localhost',
  port = 7012,
  onstream = function (sv, st)
    local rsh = headers.new()
    rsh:append(':status','200')
    rsh:append('connection','keep-alive')
    rsh:append('content-type','text/plain')
    rsh:append('content-length','5')
    st:write_headers(rsh, false)
    st:write_chunk("Hello", true)
  end
}

s:listen()
s:loop()
